src/Network.cpp

```diff
@@ -109,6 +109,12 @@ void Network::benchmark(GameState * state, int iterations) {
              iterations, elapsed, (int)(iterations / elapsed));
 }
 
+void Network::process_bn_var(std::vector<float>& weights, const float epsilon) {
+    for(size_t i=0;i < weights.size(); i++) {
```

Comment:
Should probably use a ranged based for or even replace it completely by a functional version.